code/src/App.js

```diff
@@ -1,9 +1,38 @@
-import React from 'react'
+import React, { useState } from 'react';
+import { BrowserRouter, Switch, Route, Redirect } from 'react-router-dom';
+
+import ListHeader from 'components/ListHeader';
+import Error from 'components/Error';
+import MovieList from './pages/MovieList';
+import MoviePage from './pages/MoviePage';
+import CompanyList from './pages/CompanyList';
+
+import './styles/styles.scss';
 
 export const App = () => {
+  const [list, setList] = useState();
+  const handleClick = selectedList => {
+    setList(selectedList);
+  };
+
   return (
-    <div>
-      Find me in src/app.js!
-    </div>
-  )
-}
+    <BrowserRouter>
+      <Switch>
+        <Route exact path="/">
+          <ListHeader onClick={handleClick} />
+          <MovieList>{list}</MovieList>
+        </Route>
+        <Route exact path="/movies/:movieId">
+          <MoviePage />
+        </Route>
+        <Route exact path="/company/:companyId">
+          <CompanyList />
+        </Route>
+        <Route exact path="/error">
```

Comment:
No need to add `exact` prop to every path. Take a look :

We don't need to use `exact` prop every time we specify new path. `exact` prop added to `<Route />` component specifies the URL path after which some component should be rendered. `react-router-dom` package however does the compare even without this prop. What's the difference?

Let's say our current URL in browser is `"/second"`.

In the example below, `<FirstComponent />` will be rendered instead of `<SecondComponent />`, no matter what is the browser URL path. It's because `react-router-dom` will check for common parts of url. Does `"/"` path has something in common with `"/second"` path? Yes, it's `/`. After detecting that, our package will consider search finished and render component provided for `"/"` path, so `<FirstComponent />`.

```jsx
<Switch>
  <Route path="/">
    <FirstComponent />
  </Route>
  <Route path="/second">
    <SecondComponent />
  </Route>
</Switch>
```

In the example below however, where we use `exact` prop, `<SecondComponent />` will be rendered. It's thanks to `exact`, as it says to `react-router-dom` that it should render `<FirstComponent />` only when current path is EXACTLY `"/"`. In other case, it should not be considered as the proper path.

```jsx
<Switch>
  <Route path="/" exact>
    <FirstComponent />
  </Route>
  <Route path="/second">
    <SecondComponent />
  </Route>
</Switch>
```

So, with that knowledge introduced, let's check out another example below. Over here, we don't need to use exact path for anything else than `"/"` path, because `"/second"` and `"/third"` have nothing in common in their URLs and therefore will never collide with each other. That's why we don't have to use `exact` keyword every time, but only when two paths collide with each other.

```jsx
<Switch>
  <Route path="/" exact>
    <FirstComponent />
  </Route>
  <Route path="/second">
    <SecondComponent />
  </Route>
  <Route path="/third">
    <ThirdComponent />
  </Route>
</Switch>
```